.github/workflows/testing.yaml

```diff
@@ -21,87 +21,84 @@ jobs:
-          commit_hash=$(git ls-remote https://github.com/tarantool/tarantool.git --branch ${branch} | head -c 8)
-          echo "TNT_BRANCH=${branch}" >> $GITHUB_ENV
-          echo "VERSION_POSTFIX=-${commit_hash}" >> $GITHUB_ENV
-        shell: bash
-
-      - name: Cache tarantool build
-        if: startsWith(matrix.tarantool, 'debug')
-        id: cache-tnt-debug
+      - name: Cache rocks
         uses: actions/cache@v3
+        id: cache-rocks
         with:
-          path: ${{ env.TNT_DEBUG_PATH }}
-          key: cache-tnt-${{ matrix.tarantool }}${{ env.VERSION_POSTFIX }}
+          path: .rocks/
+          key: "cache-rocks-${{ matrix.tarantool }}${{ env.VERSION_POSTFIX }}"
 
-      - name: Clone tarantool ${{ matrix.tarantool }}
-        if: startsWith(matrix.tarantool, 'debug') && steps.cache-tnt-debug.outputs.cache-hit != 'true'
-        uses: actions/checkout@v3
-        with:
-          repository: tarantool/tarantool
-          ref: ${{ env.TNT_BRANCH }}
-          path: tarantool
-          fetch-depth: 0
-          submodules: true
-
-      - name: Build tarantool ${{ matrix.tarantool }}
-        if: startsWith(matrix.tarantool, 'debug') && steps.cache-tnt-debug.outputs.cache-hit != 'true'
-        run: |
-          sudo apt-get -y install git build-essential cmake make zlib1g-dev \
-            libreadline-dev libncurses5-dev libssl-dev \
-            libunwind-dev libicu-dev python3 python3-yaml \
-            python3-six python3-gevent
-          cd ${GITHUB_WORKSPACE}/tarantool
-          mkdir build && cd build
-          cmake .. -DCMAKE_BUILD_TYPE=Debug -DENABLE_DIST=ON
-          make
-          make DESTDIR=${TNT_DEBUG_PATH} install
-
-      - name: Install tarantool ${{ matrix.tarantool }}
-        if: startsWith(matrix.tarantool, 'debug')
-        run: |
-          sudo cp -rvP ${TNT_DEBUG_PATH}/usr/local/* /usr/local/
+      - name: Install requirements
+        run: make deps depname=test
+        if: steps.cache-rocks.outputs.cache-hit != 'true'
 
+      - run: echo $PWD/.rocks/bin >> $GITHUB_PATH
+
+      - run: make test
+
+  coverage:
+    # We want to run on external PRs, but not on our own internal
+    # PRs as they'll be run by the push to the branch.
+    #
+    # The main trick is described here:
+    # https://github.com/Dart-Code/Dart-Code/pull/2375
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+
+    strategy:
+      fail-fast: false
+
+    env:
+      TNT_DEBUG_PATH: /home/runner/tnt-debug
+
+    runs-on: ubuntu-22.04
+    steps:
       - name: Clone the module
         uses: actions/checkout@v3
 
+      - name: Setup tt
+        run: |
+          curl -L https://tarantool.io/release/3/installer.sh | bash
+          sudo apt install -y tt
+          tt version
+
+      - name: Install Tarantool
+        uses: ./.github/actions/install-tarantool
+        with: 
+          tarantool: '3.1'
```

Comment:
Since you're already reworking the CI, you can bump to 3.3 as well.

Reply:
Bumped.

Reply:
Please, add a lowest supported version to a matrix to make sure that we support Tarantool since...

As example, the matrix could be:

[3.0.3, 3.3.1] - min and max known version.

Reply:
Added, thank you!

Reply:
> Please, add a lowest supported version to a test matrix to make sure that we support Tarantool since it.
> 
> As example, the matrix could be:
> 
> [3.0.3, 3.3.1] - min and max known supported version + `master`.

Why is `3.0.3` is a minimal supported version? Why not 3.0.0 for example?

Reply:
It seems that commits are mixed right now (`ci: add separate coverage job` still has old code)

Reply:
It's situated in the next commit: [ci: add the 3.3.1 tarantool verison](https://github.com/tarantool/metrics-export-role/pull/31/commits/e97d315716cdfd36a48ef8ddeee73ebc896ef6ff).